src/SFA.DAS.ApplyService.Application/Apply/Gateway/UpdateGatewayPageAnswerRequestHandler.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using MediatR;
+using System.Threading;
+using System.Threading.Tasks;
+using SFA.DAS.ApplyService.Application.Interfaces;
+using SFA.DAS.ApplyService.Domain.Audit;
+using SFA.DAS.ApplyService.Domain.Entities;
+
+namespace SFA.DAS.ApplyService.Application.Apply.Gateway
+{
+    public class UpdateGatewayPageAnswerRequestHandler : IRequestHandler<UpdateGatewayPageAnswerRequest>
+    {
+        private readonly IApplyRepository _applyRepository;
+        private readonly IAuditService _auditService;
+
+        public UpdateGatewayPageAnswerRequestHandler(IApplyRepository applyRepository, IAuditService auditService)
+        {
+            _applyRepository = applyRepository;
+            _auditService = auditService;
+        }
+
+        public async Task<Unit> Handle(UpdateGatewayPageAnswerRequest request, CancellationToken cancellationToken)
+        {
+            _auditService.StartTracking(UserAction.UpdateGatewayPageOutcome, request.UserId, request.UserName);
+
+            var application = await _applyRepository.GetApplication(request.ApplicationId);
+            var answer = await _applyRepository.GetGatewayPageAnswer(request.ApplicationId, request.PageId);
+            var isNew = answer == null;
+
+            if (answer == null)
+            {
+                answer = CreateNewGatewayPageAnswer(request.ApplicationId, request.PageId);
+                _auditService.AuditInsert(answer);
+            }
+            else
+            {
+                _auditService.AuditUpdate(answer);
+            }
+
+            _auditService.AuditUpdate(application);
```

Comment:
can you move this adjacent to the .UpdateApplication call, just easier to follow

Reply:
done